franklin/builder/tests.py

```diff
@@ -2,31 +2,135 @@
 
 from django.test import TestCase
 
-from .models import Site
+from .models import BranchBuild, Environment, Owner, Site, TagBuild
+from github.serializers import GithubWebhookSerializer
 
 
-class SitesTestCase(TestCase):
+class OwnerTestCase(TestCase):
     def setUp(self):
-        Site.objects.create(git_hash='123ABC', repo_name='example/test')
+        self.owner = Owner.objects.create(
+            name='istrategylabs', github_id=607333)
 
-    def test_site_path(self):
+    def test_owner_creation(self):
+        self.assertEqual(self.owner.name, "istrategylabs")
+
+
+class EnvironmentTestCase(TestCase):
+    def setUp(self):
+        self.owner = Owner.objects.create(
+            name='istrategylabs', github_id=607333)
+        self.site = Site.objects.create(
+            owner=self.owner, name='ampm', github_id=36470460)
+        self.env = self.site.environments.get(name=self.site.DEFAULT_ENV)
+        self.git_message = dict([('ref','refs/heads/master'),
+            ('head_commit', dict([('id', '80d25a09bc1ceb7d6707048211745dcc01bc8705')])), 
+            ('repository', dict([('id','36470460'), ('full_name','istrategylabs/ampm')]))])
+
```

Comment:
Because of the complexity of model inheritance, I decided to be TDD with this refactor. As a result, I overhauled our tests (which were broken and we didn't notice btw). In the process, I had to fix broken tests and added a bunch that mock the REST calls I was already making. 

Some of them anyway....still needs more tests. But I feel a bit better that code is guaranteed to be a bit more stable now.